### PR TITLE
add tool-types files.

### PR DESCRIPTION
**Pull request by epldlnwynn**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/tool.js/LICENSE
Comment (codershiba) [inline types/tool.js/LICENSE:1]: This LICENSE is not needed

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/tool.js/tool-js-tests.ts
1. types/tool.js/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 24d1bd554bba6cfc64889a2880ab9f04601db267
</p>

**Comment by epldlnwynn** [comment]
ok

**Comment by peterblazejewicz** [comment]
- you probably want to use global module augmentation (adding features to existing global objects):
https://www.typescriptlang.org/docs/handbook/declaration-merging.html#global-augmentation
(at the moment a string gets bare-bones features, supporting whatever your code overrides as it wins LIFO)
- this is non-nom module, but sounds strangely close to existing 'tooljs' (and similar), jus
- project ships it's own types already under `tool-types/index.d.ts`

File: types/tool.js/tool-js-tests.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import "tool-extend"
```
Comment (sheetalkamat) [inline types/tool.js/tool-js-tests.ts:1]: why ?

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
There are multiple issues already pointed out in feedback .. pls address that

**Pull request closed without merging**